spec/support/parser_examples.rb

```diff
@@ -0,0 +1,39 @@
+shared_examples_for 'a traffic source campaign parser' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.